File: 409.longest-palindrome.py
class Solution:
    def longestPalindrome(self, s: str) -> int:
        nums = {}
        for i in s:
            nums[i] = nums.get(i, 0) + 1
        answer = 0
        has_odd = False
        for count in nums.values():
            if count % 2 == 1:
                has_odd = True
                answer += count - 1
            else:
                answer += count
        return answer + (1 if has_odd else 0)

File: 409.longest-palindrome.cpp

#include <unordered_map>
class Solution {
  public:
    int longestPalindrome(string s){
		std::unordered_map<char, int> map;
		for (char c : s) {
			map[c]++;
		}
		int ans = 0;
		for (auto &p : map) {
			ans += p.second / 2 * 2;
			if (ans % 2 == 0 && p.second % 2 == 1) {
				ans++;
			}
		}
		return ans;

	} 
};

